packages/server/dataloader/foreignKeyLoaderMaker.ts

```diff
@@ -34,18 +26,20 @@ export function foreignKeyLoaderMaker<
 >(
   primaryLoaderKey: LoaderName,
   foreignKey: KeyName,
-  fetchFn: (keys: readonly LoaderType<LoaderName>[KeyName][]) => Promise<LoaderType<LoaderName>[]>
+  fetchFn: (
+    keys: readonly LoaderType<LoaderName>[KeyName][]
+  ) => Promise<(LoaderType<LoaderName> & {id: string | number})[]>
 ) {
   type T = LoaderType<LoaderName>
-  type PrimaryKeyT = LoaderKeyType<LoaderName>
+  type PrimaryKeyT = string | number
```

Comment:
The runtime logic showed that if an item was non-null, it would have an `id` field.
If that's true, then the id field is always gonna be a number or string